core/chains/evm/client/client.go

```diff
@@ -26,19 +27,16 @@ const queryTimeout = 10 * time.Second
 
 // Client is the interface used to interact with an ethereum node.
 type Client interface {
+	txmgrtypes.Client[*big.Int, common.Address, types.Block, types.Header, types.Transaction, common.Hash, types.Receipt, types.Log, ethereum.FilterQuery]
```

Comment:
does the transaction manager use all the methods in this client interface?

if it doesn't maybe it's useful to separate the Client into two parts - one that captures all the different calls, and the other one is an interface that is only used in the txm. (it helps to save injecting all the generic types that are required for other functions that the txm doesn't use)

mainly the goal would be to reduce the required interfaces that a chain-client would have to implement to be able to use the txm